### ISSUE TITLE
No resolution found at all for condition

### ISSUE THREAD

**Issue by DrClick** [systemjs/builder#584]
Hey there, I have written and angular2 app that has a structure /src/app where my main.ts file is. I am using grunt to  to copy compiled files to a .tmp folder for dev (I symlink the jspm_packages folder). I am using grunt for this. At one point I was able to bundle correctly, but now I cant. I can serve the app fine but now when I try to do this `jspm build app ../dist/main.sfx.js` I get `No resolution found at all for condition npm:reflect-metadata@0.1.3/#:crypto.` Reflect-metadata happens to be the first line of my main.ts file. I have tried this command from both the root and .tmp directory.

Here are the config files, index, main etc. Any ideas?

jspm.config.js
```
SystemJS.config({
    packages: {
        "app": {
            "main": "main.js",
            "defaultExtension": "js"
        }
    },
    map: {
        "@angular/router": "npm:@angular/router@2.0.0-rc.1"
    }
});

SystemJS.config({
    packageConfigPaths: [
        "npm:@*/*.json",
        "npm:*.json",
        "github:*/*.json"
    ],
    map: {
        "@angular/router-deprecated": "npm:@angular/router-deprecated@2.0.0-rc.1",
        "@angular/common": "npm:@angular/common@2.0.0-rc.1",
        "@angular/compiler": "npm:@angular/compiler@2.0.0-rc.1",
        "@angular/core": "npm:@angular/core@2.0.0-rc.1",
        "@angular/http": "npm:@angular/http@2.0.0-rc.0",
        "@angular/platform-browser": "npm:@angular/platform-browser@2.0.0-rc.1",
        "@angular/platform-browser-dynamic": "npm:@angular/platform-browser-dynamic@2.0.0-rc.1",
        "@angular/upgrade": "npm:@angular/upgrade@2.0.0-rc.0",
        "angular2-in-memory-web-api": "npm:angular2-in-memory-web-api@0.0.7",
        "angular2-uuid": "npm:angular2-uuid@1.0.7",
        "bootstrap": "npm:bootstrap@4.0.0-alpha.2",
        "buffer": "github:jspm/nodelibs-buffer@0.2.0-alpha",
        "css": "github:systemjs/plugin-css@0.1.21",
        "d3": "npm:d3@3.5.17",
        "es6-shim": "npm:es6-shim@0.35.0",
        "fs": "github:jspm/nodelibs-fs@0.2.0-alpha",
        "json": "github:systemjs/plugin-json@0.1.2",
        "lodash": "npm:lodash@4.11.2",
        "os": "github:jspm/nodelibs-os@0.2.0-alpha",
        "process": "github:jspm/nodelibs-process@0.2.0-alpha",
        "reflect-metadata": "npm:reflect-metadata@0.1.3",
        "rxjs": "npm:rxjs@5.0.0-beta.6",
        "systemjs": "npm:systemjs@0.19.26",
        "typescript": "npm:typescript@1.8.10",
        "zone.js": "npm:zone.js@0.6.12"
    },
    packages: {
        "github:jspm/nodelibs-buffer@0.2.0-alpha": {
            "map": {
                "buffer-browserify": "npm:buffer@4.6.0"
            }
        },
        "github:jspm/nodelibs-os@0.2.0-alpha": {
            "map": {
                "os-browserify": "npm:os-browserify@0.2.1"
            }
        },
        "npm:angular2-in-memory-web-api@0.0.7": {
            "map": {
                "@angular/core": "npm:@angular/core@2.0.0-rc.1",
                "@angular/http": "npm:@angular/http@2.0.0-rc.1",
                "es6-shim": "npm:es6-shim@0.35.0",
                "reflect-metadata": "npm:reflect-metadata@0.1.3",
                "rxjs": "npm:rxjs@5.0.0-beta.6",
                "zone.js": "npm:zone.js@0.6.12"
            }
        },
        "npm:bootstrap@4.0.0-alpha.2": {
            "map": {
                "jquery": "npm:jquery@2.2.3"
            }
        },
        "npm:buffer@4.6.0": {
            "map": {
                "base64-js": "npm:base64-js@1.1.2",
                "ieee754": "npm:ieee754@1.1.6",
                "isarray": "npm:isarray@1.0.0"
            }
        },
        "npm:systemjs@0.19.26": {
            "map": {
                "when": "npm:when@3.7.7"
            }
        }
    }
});
```

src/index
```
<!DOCTYPE html>
<html lang="en">

<head>
    <base href="/">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <title>DS Portal</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- build:css style.min.css -->
    <link rel="stylesheet" href="jspm_packages/npm/bootstrap@4.0.0-alpha.2/css/bootstrap.min.css">
    <link rel="stylesheet" href="styles/styles.css">
    <!-- /build -->

    <link rel="icon" href="favicon.png">

    <!-- build:js system.min.js -->
    <script src="node_modules/zone.js/dist/zone.js"></script>
    <script src="jspm_packages/system.js"></script>
    <script src="jspm.browser.js"></script>
    <script src="jspm.config.js"></script>
    <!-- /build -->

    <script>
        System.import('app').catch(function(err) {
                console.error(err);
            });
    </script>
</head>

<!-- 3. Display the application -->

<body>
    <my-app>
        <loading></loading>
    </my-app>
</body>

</html>
```

src/app/main
```
import 'reflect-metadata';


import {bootstrap} from '@angular/platform-browser-dynamic';
import {AppComponent} from './app.component';
import {ROUTER_PROVIDERS} from '@angular/router-deprecated';
import {AuthService} from "./shared/auth.service";
import {HTTP_PROVIDERS, Http} from "@angular/http";
import {provide} from "@angular/core";
import {environment} from './config/app.config';

//The enviornment (app.config) config will be copied from the grunt build process from the root config folders
//to this central location based on the build type.

//test this again
let ConfigProvider = provide('CONFIG', {useValue: environment});
bootstrap(AppComponent, [ROUTER_PROVIDERS, HTTP_PROVIDERS, AuthService, ConfigProvider])
```

**Comment by j-walker23** [comment]
@DrClick quick question for you. I just started having this problem on the latest jspm@beta. Since i wasn't on the previous versions i tried to re-install those and no matter what i get the same error.

**Comment by guybedford** [comment]
I can confirm that this is working with the latest conditional updates in SystemJS builder coming in the next release.

**Comment by guybedford** [comment]
Released in 0.15.17.

**Comment by j-walker23** [comment]
@guybedford Thank you! Also, using `jspm 0.17` has been a great experience. Thank you for that as well!
